### ISSUE TITLE
Crash in SDK version 2.7.0 - NotSerializableException: android.net.Uri$StringUri

### ISSUE THREAD

**Issue by LasseMortensen** [Nets-mobile-acceptance/Netaxept-Android-SDK#71]
**SDK version:** 2.7.0
**Android Versions:** 100% - Android 11
**Frequency:** 100% when using saved payment card

**Stacktrace**
```
FATAL EXCEPTION: main
Process: com.nemlig.test, PID: 6863
java.lang.RuntimeException: Parcelable encountered IOException writing serializable object (name = eu.nets.pia.CardProcessError$TerminalRequest)
	at android.os.Parcel.writeSerializable(Parcel.java:2125)
	at eu.nets.pia.ProcessResult$Failure.writeToParcel(Unknown Source:12)
	at android.os.Parcel.writeParcelable(Parcel.java:1916)
	at android.os.Parcel.writeValue(Parcel.java:1822)
	at android.os.Parcel.writeArrayMapInternal(Parcel.java:987)
	at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1620)
	at android.os.Bundle.writeToParcel(Bundle.java:1303)
	at android.os.Parcel.writeBundle(Parcel.java:1056)
	at android.content.Intent.writeToParcel(Intent.java:11512)
	at android.app.IActivityTaskManager$Stub$Proxy.finishActivity(IActivityTaskManager.java:4840)
	at android.app.Activity.finish(Activity.java:6452)
	at android.app.Activity.finish(Activity.java:6488)
	at a.a.a.d.a.b.a(Unknown Source:76)
	at eu.nets.pia.ui.main.PiaActivity.a(Unknown Source:13)
	at a.a.a.d.e.d$a.run(Unknown Source:98)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:246)
	at android.app.ActivityThread.main(ActivityThread.java:8633)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
Caused by: java.io.NotSerializableException: android.net.Uri$StringUri
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1240)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1604)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1565)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1488)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1234)
	at java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1604)
	at java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1565)
	at java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1488)
	at java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1234)
	at java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:354)
	at android.os.Parcel.writeSerializable(Parcel.java:2120)
	at eu.nets.pia.ProcessResult$Failure.writeToParcel(Unknown Source:12) 
	at android.os.Parcel.writeParcelable(Parcel.java:1916) 
	at android.os.Parcel.writeValue(Parcel.java:1822) 
	at android.os.Parcel.writeArrayMapInternal(Parcel.java:987) 
	at android.os.BaseBundle.writeToParcelInner(BaseBundle.java:1620) 
	at android.os.Bundle.writeToParcel(Bundle.java:1303) 
	at android.os.Parcel.writeBundle(Parcel.java:1056) 
	at android.content.Intent.writeToParcel(Intent.java:11512) 
	at android.app.IActivityTaskManager$Stub$Proxy.finishActivity(IActivityTaskManager.java:4840) 
	at android.app.Activity.finish(Activity.java:6452) 
	at android.app.Activity.finish(Activity.java:6488) 
	at a.a.a.d.a.b.a(Unknown Source:76) 
	at eu.nets.pia.ui.main.PiaActivity.a(Unknown Source:13) 
	at a.a.a.d.e.d$a.run(Unknown Source:98) 
	at android.os.Handler.handleCallback(Handler.java:938) 
	at android.os.Handler.dispatchMessage(Handler.java:99) 
	at android.os.Looper.loop(Looper.java:246) 
	at android.app.ActivityThread.main(ActivityThread.java:8633) 
	at java.lang.reflect.Method.invoke(Native Method) 
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602) 
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130) 
win=Window{b4ba7ca u0 com.nemlig.test/eu.nets.pia.ui.main.PiaActivity EXITING} destroySurfaces: appStopped=false win.mWindowRemovalAllowed=true win.mRemoveOnExit=true win.mViewVisibility=0 caller=com.android.server.wm.ActivityRecord.destroySurfaces:5251 com.android.server.wm.ActivityRecord.destroySurfaces:5232 com.android.server.wm.WindowState.onExitAnimationDone:5623 com.android.server.wm.WindowStateAnimator.onAnimationFinished:332 com.android.server.wm.WindowState.onAnimationFinished:6088 com.android.server.wm.-$$Lambda$dwJG8BAnLlvKNGuDY9U3-haNY4M.onAnimationFinished:2 com.android.server.wm.SurfaceAnimator.lambda$getFinishedCallback$0$SurfaceAnimator:112 
```

**Steps to reproduce**
Use saved payment card for purchase. Can be reproduced with test card 4925000000000004

**Comment by Miss-PiA** [comment]
Hi @LasseMortensen ,
Sorry for this inconvenience. We will look into this issue and update you soon.
 
Thanks,

**Comment by Miss-PiA** [comment]
Hi @LasseMortensen 

I'm unable to reproduce the issue. I tried both in debug and release mode. Are you able to reproduce the issue in our [sample application](https://github.com/Nets-mobile-acceptance/Netaxept-Android-SDK/tree/master/PiaSample), if you are able to do so, can you please let me know the steps?

**Comment by Miss-PiA** [comment]
Hi @LasseMortensen 

Can you please provide merchant ID and transaction ID for a failing transaction? That information will be quite helpful for us

**Comment by Miss-PiA** [comment]
Hi @LasseMortensen 

Can you please check if your issue has been resolved in v2.7.1 release of the SDK?

**Comment by BuildManager-Nets** [comment]
Hi @LasseMortensen,

Currently closing the issue. 
In case you are facing this issue again in the latest version please reopen the issue.

Thanks
